cookbook/security/multiple_guard_authenticators.rst

```diff
@@ -0,0 +1,168 @@
+
+Multiple authenticators with shared entry point
+-----------------------------------------------
+
+Let's have an example of two authenticators: one based on login form, another one on facebook login.
+Both authenticators entry points redirect user to the same login page.
+However, in your configuration you have to explicitly say which entry point you want to use.
+
+This is how your security configuration can look in action:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/security.yml
+        security:
+             # ...
+            firewalls:
+                default:
+                    anonymous: ~
+                    guard:
+                        authenticators:
+                            - app.form_login_authenticator
+                            - app.facebook_connect_authenticator
+                        entry_point: app.form_login_authenticator
+
+    .. code-block:: xml
+
+        <!-- app/config/security.xml -->
+        <?xml version="1.0" encoding="UTF-8"?>
+        <srv:container xmlns="http://symfony.com/schema/dic/security"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:srv="http://symfony.com/schema/dic/services"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <config>
+                <!-- ... -->
+                <firewall name="default">
+                    <anonymous />
+                    <guard entry_point="app.form_login_authenticator">
+                        <authenticator>app.form_login_authenticator</authenticator>
+                        <authenticator>app.facebook_connect_authenticator</authenticator>
+                    </guard>
+                </firewall>
+            </config>
+        </srv:container>
+
+    .. code-block:: php
+
+        // app/config/security.php
+        $container->loadFromExtension('security', array(
+            // ...
+            'firewalls' => array(
+                'default' => array(
+                    'anonymous' => null,
+                    'guard' => array(
+                        'entry_point' => 'app.form_login_authenticator',
+                        'authenticators' => array(
+                            'app.form_login_authenticator',
+                            'app.facebook_connect_authenticator'
+                        ),
+                    ),
+                ),
+            ),
+        ));
+
+There is one limitation with this approach - you have to use exactly one entry point.
+
+Multiple authenticators with separate entry points
+--------------------------------------------------
+
+Let's now have an example of two different authenticators: one based on login form, another one on an API token.
+When user hits secured area he should be redirected to the login page.
+Also when user hits an API endpoint, he should get a relevant API response.
```

Comment:
I would replace "Also when user hits[...]" with "But when a user hits [...]". We also use the gender-neutral form in the docs, please replace "he should get a relevant API response" with "they should get [...]".